framework/src/engine/generator/generator.ts

```diff
@@ -199,9 +199,12 @@ export class Generator {
 		});
 
 		const stateStore = new StateStore(this._blockchainDB);
-		const maxRemovalHeight = await this._consensus.getMaxRemovalHeight();
-		const { maxHeightPrecommitted } = await this._bft.method.getBFTHeights(stateStore);
-		await Promise.all(this._handleFinalizedHeightChanged(maxRemovalHeight, maxHeightPrecommitted));
+		const { maxHeightPrecommitted, maxHeightCertified } = await this._bft.method.getBFTHeights(
+			stateStore,
+		);
+		await Promise.all(
```

Comment:
As this was a point of confusion before, maybe we could add some explanation to the code as a comment, e.g., 
"A certificate for height maxHeightCertified is already generated, hence it is only necessary to create single commits for height maxHeightCertified + 1 onward".

Reply:
Or we link to the specific section of the LIP here, i.e., https://github.com/LiskHQ/lips/blob/main/proposals/lip-0061.md#initial-single-commit-creation
(I need to fix it and replace `h1 = getMaxRemovalHeight()` by `h1 = maxHeightCertified`.

Reply:
I created an issue for it https://github.com/LiskHQ/lips/issues/245.